packages/domain-lib/test/unit/index.test.ts

```diff
@@ -1288,16 +1288,229 @@ describe("Settlements BC [Domain] - Unit Tests", () => {
 		expect(awaitSettleByBatch!.state).toEqual('AWAITING_SETTLEMENT');
 
 		// batch is allowed to be added to the matrix, but not allowed to lock:
-		const matrixNoLock = await aggregate.getSettlementMatrix(securityContext, matrixIdNoLock);
+		let matrixNoLock = await aggregate.getSettlementMatrix(securityContext, matrixIdNoLock);
+		expect(matrixNoLock).toBeDefined();
+		expect(matrixNoLock!.id).toEqual(matrixIdNoLock);
+		expect(matrixNoLock!.state).toEqual('IDLE');
+		expect(matrixNoLock!.type).toEqual('STATIC');
+		expect(matrixNoLock!.participantBalances.length).toEqual(2);
+		expect(matrixNoLock!.batches.length).toEqual(1);
+
+		await aggregate.settleSettlementMatrix(securityContext, matrixIdNoLock);
+		matrixNoLock = await aggregate.getSettlementMatrix(securityContext, matrixIdNoLock);
 		expect(matrixNoLock).toBeDefined();
 		expect(matrixNoLock!.id).toEqual(matrixIdNoLock);
-		expect(matrixNoLock!.state).toEqual('AWAITING_SETTLEMENT');
+		expect(matrixNoLock!.state).toEqual('FINALIZED');
 		expect(matrixNoLock!.type).toEqual('STATIC');
 		expect(matrixNoLock!.participantBalances.length).toEqual(2);
 		expect(matrixNoLock!.batches.length).toEqual(1);
 	});
 
-	test("test matrix state machine", async () => {
-		// TODO
+	test("test matrix state machine - positive", async () => {
+		const reqTransferDto: ITransferDto = {
+			id: null,
+			transferId: randomUUID(),
+			payerFspId: randomUUID(),
+			payeeFspId: randomUUID(),
+			currencyCode: 'EUR',
+			amount: '10000', //100 EURO
+			timestamp: Date.now(),
+			settlementModel: 'STATE_MACHINE'
+		};
+		const batchId: string = await aggregate.handleTransfer(securityContext, reqTransferDto);
+		expect(batchId).toBeDefined();
+
+		// Initial state for a batch is [OPEN]:
+		let batch = await settleBatchRepo.getBatch(batchId);
+		expect(batch).toBeDefined();
+		expect(batch!.id).toEqual(batchId);
+		expect(batch!.state).toEqual('OPEN');
+
+		// create a settlement matrix with batches:
+		let matrixId = await aggregate.createStaticSettlementMatrix(
+			securityContext,
+			null, // matrix-id
+			[batchId]
+		);
+		expect(matrixId).toBeDefined();
+
+		// close:
+		await aggregate.closeSettlementMatrix(securityContext, matrixId);
+		batch = await settleBatchRepo.getBatch(batchId);
+		expect(batch).toBeDefined();
+		expect(batch!.id).toEqual(batchId);
+		expect(batch!.state).toEqual('CLOSED');
```

Comment:
Then we can show here (after ln1342) that once the batch state is `CLOSED` then no more transactions can be added to the batch.

Reply:
Fixed.